src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AttendanceService.kt

```diff
@@ -329,6 +332,46 @@ class AttendanceService(
       }.toSet()
   }
 
+  private fun addAttendance(
+    map: MutableMap<YearMonth, AttendanceSummary>,
+    offenderAttendance: OffenderAttendance
+  ): MutableMap<YearMonth, AttendanceSummary> {
+
+    val yearMonth = YearMonth.of(offenderAttendance.eventDate.year, offenderAttendance.eventDate.month)
+    var slot = map.get(yearMonth)
+    if (slot == null) {
+      val new = AttendanceSummary(yearMonth)
+      map.put(yearMonth, new)
+      slot = new
+    }
```

Comment:
think you might be able to do? 

`val slot = map.getOrPut(yearMonth) { AttendanceSummary(yearMonth) }`